src/apps/MASS3DPA-Seq.cpp

```diff
@@ -0,0 +1,195 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "MASS3DPA.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#include <iostream>
+
+namespace rajaperf {
+namespace apps {
+
+#define RAJA_UNROLL(N)
+#define FOREACH_THREAD(i, k, N) for (int i = 0; i < N; i++)
+
+void MASS3DPA::runSeqVariant(VariantID vid) {
+  const Index_type run_reps = getRunReps();
+
+  MASS3DPA_DATA_SETUP;
+
+  switch (vid) {
+
+  case Base_Seq: {
+
+    startTimer();
+    for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+
+      for (int e = 0; e < NE; ++e) {
+
+        MASS3DPA_0_CPU
+
+        FOREACH_THREAD(dy, y, D1D) {
+          FOREACH_THREAD(dx, x, D1D){MASS3DPA_1}
+          FOREACH_THREAD(dx, x, Q1D) {MASS3DPA_2}
+        }
+
+        FOREACH_THREAD(dy, y, D1D) {
+          FOREACH_THREAD(qx, x, Q1D) { MASS3DPA_3 }
+        }
+
+        FOREACH_THREAD(qy, y, Q1D) {
+          FOREACH_THREAD(qx, x, Q1D) { MASS3DPA_4 }
+        }
+
+        FOREACH_THREAD(qy, y, Q1D) {
+          FOREACH_THREAD(qx, x, Q1D) { MASS3DPA_5 }
+        }
+
+        FOREACH_THREAD(d, y, D1D) {
+          FOREACH_THREAD(q, x, Q1D) { MASS3DPA_6 }
+        }
+
+        FOREACH_THREAD(qy, y, Q1D) {
+          FOREACH_THREAD(dx, x, D1D) { MASS3DPA_7 }
+        }
+
+        FOREACH_THREAD(dy, y, D1D) {
+          FOREACH_THREAD(dx, x, D1D) { MASS3DPA_8 }
+        }
+
+        FOREACH_THREAD(dy, y, D1D) {
+          FOREACH_THREAD(dx, x, D1D) { MASS3DPA_9 }
+        }
+
+      } // element loop
+    }
+    stopTimer();
+
+    break;
+  }
+
+#if defined(RUN_RAJA_SEQ)
+  case RAJA_Seq: {
+
+    //Currently Teams requires two policies if compiled with a device
+    using launch_policy = RAJA::expt::LaunchPolicy<RAJA::expt::seq_launch_t
+#if defined(RAJA_DEVICE_ACTIVE)
```

Comment:
Are these macro guarded code sections here for consistency with the GPU variants?

Reply:
It's in case we compile with a device; at the moment we force having to provide both (host/device) policies. Future work in Launch/Teams will rework the framework so its not forced.

Reply:
Please add a GitHub issue to address that in the kernel implementation when it's supported in RAJA.